### PR TITLE
SQS - Fix message attributes disappearing on multiple receives

### PR DESCRIPTION
**Pull request by dominikschubert**
See the test for the behavior this PR is trying to fix.

Because the filtering in `receive_message` was removing items from the attributes on the actual message item in the backend, the next (less restricted) call would not be able to get the previously filtered out attributes anymore.

### CODE REVIEW

File: moto/sqs/models.py [hunk 1]
```diff
@@ -534,7 +534,12 @@ def add_message(self, message):
                 [backend.delete_message(self.name, m.receipt_handle) for m in messages]
             else:
                 # Make messages visible again
-                [m.change_visibility(visibility_timeout=0) for m in messages]
+                [
```
Comment (bblommers) [inline moto/sqs/models.py:537]: What's the reason for this change?

File: tests/test_sqs/test_sqs.py [hunk 2]
```diff
@@ -3315,3 +3315,43 @@ def test_message_attributes_contains_trace_header():
...
+            "Custom1": {"StringValue": "Custom_Value_1", "DataType": "String"},
+            "Custom2": {"StringValue": "Custom_Value_2", "DataType": "String"},
+        },
+    )
+
+    first_messages = conn.receive_message(
+        QueueUrl=queue.url,
+        MaxNumberOfMessages=2,
+        MessageAttributeNames=["Custom1"],
+        VisibilityTimeout=1,
```
Comment (bblommers) [inline tests/test_sqs/test_sqs.py:3341]: If the VisibilityTimeout is 0, we shouldn't need the `sleep`-statement anymore - would help a lot in speeding up the tests

**Review by bblommers — CHANGES_REQUESTED** [review]

File: moto/sqs/models.py (hunk 1 above)
Comment (dominikschubert) [reply moto/sqs/models.py:537]: Otherwise it could now modify the returned message copy (the one with possibly filtered attributes) and the original message in the backend wouldn't be touched.

By using `backend.change_message_visibility` this can be avoided since it only references the receipt handle of the message and will do a lookup of the message in its internal `_messages` field. Previously this worked because these were always the same message objects.


This change was now required for this test to pass after the changes of this PR: 

https://github.com/spulec/moto/blob/be197caba6282260ca53112f1f5c7ab4a6675ca9/tests/test_awslambda/test_lambda_eventsourcemapping.py#L213-L274

File: tests/test_sqs/test_sqs.py (hunk 2 above)
Comment (dominikschubert) [reply tests/test_sqs/test_sqs.py]: Good point, I've updated the test :+1:

File: moto/sqs/models.py (hunk 1 above)
Comment (bblommers) [reply moto/sqs/models.py:537]: Gotcha, that makes sense

**Review by bblommers — APPROVED** [review]
LGTM! Thanks for the PR @dominikschubert
